### PR TITLE
EEG Device Pairing

### PR DESCRIPTION
**Pull request by Supereg**
# EEG Device Pairing

## :recycle: Current situation & Problem
As of right now the app doesn't integrate any EEG devices to connect to and perform EEG measurements.

## :bulb: Proposed solution
This PR adds an initial version of integration with Muse EEG devices for pairing. Particularly we add a `Nearby Devices` view to discovery and pair with nearby Muse EEG headbands and view the device details of a currently connected headband.
Additionally, we add a very basic view to explore the live EEG measurement stream of the currently connected Muse headband.

Below are some visual illustrations of the new functionality:
<p float="left">
Nearby Devices Screen. One device connected.
Detailed Information of the currently paired headband.
EEG Recording screen show four separate charts for all four electrodes of the Muse Headband.
</p>

The Muse SDK is available as invite only (see [here](https://choosemuse.my.site.com/s/article/Muse-Software-Development-Kit-SDK-FAQs?language=en_US)). Therefore, the SDK itself is pulled in as a git submodule.

As the SDK only compiles for iOS and not for the iOS simulator, the Xcode project adds a new `NAMS Muse` target. The `NAMS` target uses a Mock Device Layer for previewing and testing while `NAMS Muse` includes the live Muse SDK used to deploy the app.

## :gear: Release Notes 
* Integrate Muse EEG Headband devices

### Related PRs
--

### Testing
Unit tests were added based on the mock device implementation. The Muse code cannot be tested in the CI setup as of right now.

### Reviewer Nudging
It might make sense to visually navigate the new UI and, from that point, review the device layer implementation. It makes sense to review the device layer in conjunction with the Muse SDK implementation as this was one of the driving factors for the thin abstraction layer.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PSchmiedmayer** [comment]
@Supereg As discussed in the last meeting: I took a look at the fastlane setup and changed it so the test are run on the main target while releases are created using the Muse target. Also updated some other elements in the scheme to allow testing for the different schemes using the test plan.

I have seen some failing builds due to the settings app crashing in the simulator and just restarted our VM that host the builds here. Hope this will fix this.

**Comment by PSchmiedmayer** [comment]
@Supereg Let me know if you continuously run into build errors due to our build agents.

**Comment by Supereg** [comment]
> @Supereg Let me know if you continuously run into build errors due to our build agents.

Seems like there were some issues with the Schemes. Xcode seemingly changed them itself? Or I messed up? Whatever, it works now. And I disabled CodeQL as the PR bumps the deployment target to iOS 17 due to new API usage (see https://github.com/StanfordSpezi/SpeziTemplateApplication/issues/43).

**Comment by Supereg** [comment]
@PSchmiedmayer I'm fighting a bit with code coverage right now. It seems like codecov suddenly counts all previews as "Not covered" where they previously were seemingly excluded from code coverage. Do you have any experience or ideas where this might stem from?
Apart from that, I would propose to ignore any decrease that comes from the not-covered Muse code.

File: .github/workflows/beta-deployment.yml [hunk 1]
```diff
@@ -15,17 +15,18 @@ on:
...
 
 jobs:
-  buildandtest:
-    name: Build and Test
-    uses: ./.github/workflows/build-and-test.yml
-    secrets: inherit
+  # buildandtest:
+  #   name: Build and Test
+  #   uses: ./.github/workflows/build-and-test.yml
+  #   secrets: inherit
```
Comment (Supereg) [inline .github/workflows/beta-deployment.yml:21]: @PSchmiedmayer can I remove this Job completely, or what was the reason to uncomment that?

File: fastlane/Fastfile [hunk 2]
```diff
@@ -53,21 +74,22 @@ platform :ios do
...
+    #   distribute_external: true,
+    #   groups: [
+    #     "External Testers"
+    #   ],
+    #   submit_beta_review: true,
+    #   notify_external_testers: true,
+    #   expire_previous_builds: true,
+    #   changelog: commit[:message]
+    # )
   end
```
Comment (Supereg) [inline fastlane/Fastfile:94]: @PSchmiedmayer Also, the actual upload to TestFlight is not yet enabled? Can I just enable that or do I have to consider anything there?

**Comment by PSchmiedmayer** [comment]
Nice job @Supereg! 🚀

**Pull request merged**